sips/sip-011/sip-011_ustx-marketplace-standard.md

```diff
@@ -0,0 +1,427 @@
+# Preamble
+
+SIP Number: 011
+
+Title: The buy-in-ustx Standard for non-custodial Marketplace traits
+
+Authors: werner.btc (werner at stx.fan), Friedger, Mike, Jason and Jamil.
+
+Consideration: Technical
+
+Type: Standard
+
+Status: draft
+
+Created: 31 December 2022
+
+License: CC0-1.0
+
+Sign-off: Brice Dobry <brice@hiro.so>, Technical Consideration Advisory Boards (CAB) Chair
+Sign-off: Jude Nelson <jude@stacks.org>, Technical Steering Committee Chair
+
+# Abstract
+
+This standard aims to formalize a the commonly used traits to offer fungible or non-fungible tokens for sale, at a certain price in STX, from a non-custodial wallet.
+
+# License and Copyright
+
+This SIP is made available under the terms of the Creative Commons CC0 1.0 Universal license, available at https://creativecommons.org/publicdomain/zero/1.0/
+This SIP’s copyright is held by the Stacks Open Internet Foundation.
+
+# Introduction
+
+Tokens are digital assets registered on a blockchain through a smart contract. A non-fungible token (NFT) is a token that is globally unique and can be identified through its unique identifier.
+
+In blockchains with smart contracts, including the Stacks blockchain, developers and users can use smart contracts to register and interact with (non-)fungible tokens.
+
+To extend the functions of tokens this SIP defines two traits with smart contract functions that enable an open, decentralized marketplace for digital assets. These assets must implement the `marketplace trait` and `commission trait`. Buyers should be able to buy the assets from the owner with STX.
+
+The `marketplace trait` can be implemented as part of the asset contract (usually not recommended) or as an independent contract.
+
+The marketplace functions have been used since the first Megapont contract was deployed to mainnet in October 2022 and have since become the pseudo standard for non-custodial listing of assets on the Stacks Blockchain. Having this ratified as an official standard can help further adoption of the standard. [(1)](https://) [(2)](https://github.com/stacksgov/sips/issues/92#issuecomment-1360144565). A more generalized trait is also recommended to replace this one in the future. A standard that will be compatible with SIP013 tokens too. And one that will allow buying with SIP013 or SIP010 tokens aswell as STX. 
+
+# Specification
+
+Every SIP-011 compliant smart contract on the Stacks blockchain must implement these traits;
+1. `marketplace-functions`, defined in the [Marketplace-Trait-and-Commision-Trait](#Marketplace-Trait-and-Commision-Trait) section,
+2. `commission-functions`, defined in the [Marketplace-Trait-and-Commision-Trait](#Marketplace-Trait-and-Commision-Trait) section;
+
+Optionally seperate out royalties, as in the reference?
+3. See from line 440
+
+### Marketplace-Trait function 1: List in ustx
```

Comment:
Can you add an H2 header here called `Marketplace trait` that groups its functions inside? You're already using H3s for each function so those are OK.

You can then adjust the function titles to only have function names e.g. `### List in ustx`